### PR TITLE
Increase unit test coverage

### PR DESCRIPTION
**Pull request by atye**
# Description

Increase unit test coverage to satisfy 80%.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1559 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Built the powerstore image pulling the latest changes and sanity run, and tests are passing.
[root@ cert-csi]# ./cert-csi test scaling --sc powerstore
[2024-11-21 06:33:17]  INFO Starting cert-csi; ver. 1.6.0
[2024-11-21 06:33:17]  INFO Using EVENT observer type
[2024-11-21 06:33:17]  INFO Using config from /etc/kubernetes/admin.conf
[2024-11-21 06:33:17]  INFO Successfully loaded config. Host: https://X.X.X.X:6443
[2024-11-21 06:33:17]  INFO Created new KubeClient
[2024-11-21 06:33:17]  INFO Running 1 iteration(s)
[2024-11-21 06:33:17]  INFO     *** ITERATION NUMBER 1 ***
[2024-11-21 06:33:17]  INFO Starting ScalingSuite with powerstore storage class
[2024-11-21 06:33:17]  INFO Successfully created namespace scale-test-5ae9a3bb
[2024-11-21 06:33:17]  INFO Using default volume size 3Gi
[2024-11-21 06:33:17]  INFO Using default image: quay.io/centos/centos:latest
[2024-11-21 06:33:17]  INFO Created stateful set sts-scale-test68np7
[2024-11-21 06:33:17]  INFO Waiting for sts pods to become ready
[2024-11-21 06:33:27]  INFO Scaling to 5 replicas
[2024-11-21 06:33:27]  INFO Waiting for sts pods to become ready
[2024-11-21 06:34:01]  INFO Scaling to 0 replicas
[2024-11-21 06:34:01]  INFO Waiting for sts pods to become ready
[2024-11-21 06:34:03]  INFO Deleting all resources in namespace scale-test-5ae9a3bb
[2024-11-21 06:35:06]  INFO Namespace scale-test-5ae9a3bb was deleted in 1m2.761504512s
[2024-11-21 06:35:07]  INFO SUCCESS: ScalingSuite in 1m50.080913067s
[2024-11-21 06:35:07]  INFO Started generating reports...
Collecting metrics
1 / 1 [----------------------------------------------------------------------------------------------------] 100.00% ? p/s[2024-11-21 06:35:07]  INFO Started generating reports...
Collecting metrics
1 / 1 [----------------------------------------------------------------------------------------------------] 100.00% ? p/sGenerating plots
1 / 1 [----------------------------------------------------------------------------------------------------] 100.00% ? p/s[2024-11-21 06:35:07]  WARN No ResourceUsageMetrics provided
[2024-11-21 06:35:07] ERROR no ResourceUsageMetrics provided
report-test-run-f57749d9:
Name: test-run-f57749d9
Host: https://10.x.x.x:6443
StorageClass: powerstore
Minimum and Maximum EntityOverTime charts:

/root/.cert-csi/reports/test-run-f57749d9/PodsCreatingOverTime.png

/root/.cert-csi/reports/test-run-f57749d9/PodsReadyOverTime.png

/root/.cert-csi/reports/test-run-f57749d9/PodsTerminatingOverTime.png

/root/.cert-csi/reports/test-run-f57749d9/PvcsCreatingOverTime.png

/root/.cert-csi/reports/test-run-f57749d9/PvcsBoundOverTime.png

Tests:
--------------------------------------------------------------
1. TestCase: ScalingSuite
            Started:   2024-11-21 06:33:17.28866488 +0000 UTC
            Ended:     2024-11-21 06:35:07.369374214 +0000 UTC
            Result:    SUCCESS

            Stage metrics:
                    PVCAttachment:
                        Avg: 11.122264945s
                        Min: 2.203834177s
                        Max: 18.315703549s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCAttachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCAttachment_boxplot.png
                    PVCBind:
                        Avg: 6.601522997s
                        Min: 687.94461ms
                        Max: 11.953379575s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCBind.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCBind_boxplot.png
                    PVCCreation:
                        Avg: 20.689414693s
                        Min: 5.380588423s
                        Max: 30.885137696s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCCreation_boxplot.png
                    PVCDeletion:
                        Avg: 1.189982465s
                        Min: 16.819411ms
                        Max: 3.313582897s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCDeletion_boxplot.png
                    PVCUnattachment:
                        Avg: 8.291617937s
                        Min: 549.746206ms
                        Max: 16.798273442s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCUnattachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PVCUnattachment_boxplot.png
                    PodCreation:
                        Avg: 23.73764673s
                        Min: 8.428755469s
                        Max: 32.15308464s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PodCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PodCreation_boxplot.png
                    PodDeletion:
                        Avg: 1.478875431s
                        Min: 857.031716ms
                        Max: 1.862356447s
                        Histogram:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PodDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/PodDeletion_boxplot.png
                        EntityNumberOverTime:
        /root/.cert-csi/reports/test-run-f57749d9/ScalingSuite8/EntityNumberOverTime.png

[2024-11-21 06:35:07]  INFO Avg time of a run:   46.05s
[2024-11-21 06:35:07]  INFO Avg time of a del:   62.76s
[2024-11-21 06:35:07]  INFO Avg time of all:     110.08s
[2024-11-21 06:35:07]  INFO During this run 100.0% of suites succeeded
[root@ cert-csi]#

### CODE REVIEW

**Review by AkshaySainiDell — APPROVED** [review]

**Review by suryagupta4 — APPROVED** [review]

**Review by harshitap26 — DISMISSED** [review]

**Review by bpjain2004 — DISMISSED** [review]

**Review by harishp8889 — DISMISSED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -22,7 +23,8 @@
 # Santhosh shekarappa (santhoshhs10)
 # Yamunadevi Shanmugam (shanmydell)
 # Sushma T S (tssushma)
- 
+# Shayna Finocchiaro (shaynafinocchiaro)
+# Luna Xu (xuluna) 
 
 # for all files:
-* @boyamurthy @bpjain2004 @coulof @delldubey @francis-nijay @harishp8889 @hoppea2 @kumarp20 @nidtara @prablr79 @tdawe @santhoshhs10 @shanmydell @tssushma
+* @boyamurthy @bpjain2004 @coulof @delldubey @francis-nijay @harishp8889 @hoppea2 @kumarp20 @nidtara @prablr79 @tdawe @santhoshhs10 @shanmydell @tssushma @alikdell @atye @shaynafinocchiaro @xuluna
```
Comment (adarsh-dell) [inline .github/CODEOWNERS:30]: See if we can list these name in alphabetical orders.

File: Testing/gonvme_utils.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ * Copyright © 2022-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (adarsh-dell) [inline Testing/gonvme_utils.go:3]: As we are introducing this file in 2024, why 2022?

File: Testing/gonvme_utils_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
+package gonvme
```
Comment (adarsh-dell) [inline Testing/gonvme_utils_test.go:1]: Do we need : Apache License text here too?

File: Testing_gonvme/gonvme.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ * Copyright © 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (adarsh-dell) [inline Testing_gonvme/gonvme.go:3]: 2024 year?
Same hold for other files as well.

File: gonvme_tcp_fc.go [hunk 5]
```diff
@@ -811,13 +822,44 @@ func (nvme *NVMe) GetNVMeDeviceData(path string) (string, string, error) {
...
+
+	/*
+		[
+		  {
+		    "HostNQN":"nqn.2014-08.org.nvmexpress:uuid:6f08058a-af91-46bf-8311-a60da3a10348",
+		    "HostID":"6f08058a-af91-46bf-8311-a60da3a10348",
+		    "Subsystems":[
+		      {
+		        "Name":"nvme-subsys0",
+		        "NQN":"nqn.1988-11.com.dell:powerstore:00:1b7322d7546dFD05675D",
```
Comment (adarsh-dell) [inline gonvme_tcp_fc.go:841]: I can see lots of commented code, do we really need them?

**Comment by karthikk92** [comment]
Will add Single PR for Both defect fix and UT coverage

**Pull request closed without merging**